sourcecode/npm/components/src/components/AuthWrapper.jsx

```diff
@@ -48,6 +48,17 @@ const AuthWrapper = ({ children }) => {
         store.set('language', language);
     }, [firstName, lastName, email, id, language]);
 
+    const [firstNameErrorText, setfirstNameErrorText] = React.useState("");
+    const [lastNameErrorText, setlastNameErrorText] = React.useState("");
+    const [emailErrorText, setemailErrorText] = React.useState("");
+    const [idErrorText, setidErrorText] = React.useState("");
+    const validateEmail = (email) => {
+        return !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
```

Comment:
This regular expression is too strict, a valid email address like `foo@example.brussels` wouldn't pass due to the limit on top-level domain characters.

Domain name rules are always changing, and proper email validation is extremely complicated, so it's best to just stick to something like `/^[^@]+@([^@]+\.)*[^@.]+$/`.